tracks/elm/exercises/leap/mentoring.md

```diff
@@ -0,0 +1,90 @@
+This is a good exercise to discuss the describability of logic, as most students solve the problem using a conditional statement that is hard to describe in language, and there are other options.
+
+This exercise introduces `Basics.remainderBy`, the `let` / `in` syntax and partial application / currying.
+
+### Reasonable solutions
+
+The easiest / most common solution probably involves partially applying `Basics.remainderBy` and using a slightly complicated if statement:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+        (
+            divisibleBy 4 && 
+            not (divisibleBy 100)
+        )
+        || divisibleBy 400
+```
+
+The complicated if statement is a bit of a problem, as it is hard to describe in language, and hence talking about the code with others is difficult.
+
+Using a case statement turns the if statement in to a truth table, and is probably a bit clearer, but at the expense of some extra code:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+		case ( divisibleBy 4, divisibleBy 100, divisibleBy 400 ) of
+			( True, True, True ) ->
+				True
+
+			( True, True, False ) ->
+				False
+
+			( True, False, False ) ->
+				True
+
+			_ ->
+				False
+```
+
+If you think about the easiest way to describe the logic to somebody else, you probably come up with a solution like this:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+		if divisibleBy 400 then
+			True
+
+		else if divisibleBy 100 then
+			False
+
+		else if divisibleBy 4 then
+			True
+
+		else
+			False
+```
+
+It is possible to go even further with the describability of the logic, using code like this:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+	let
+		...
+	in
+		if is400YearSpecialCase then
+			True
+
+		else if is100YearSpecialCase then
+			False
+
+		else 
+			defaultCase
```

Comment:
Correct me if I'm wrong: I presume the point of this example is to show that you can name things very particularly to add meaning to some nested expression. If this is the case, I agree very much with the sentiment, which I think is also embedded in recommending a helper function such as `divisibleBy`.

But I think this particular example is not a good one, since `is400YearSpecialCase` doesn't say as much as `divisibleBy 400`. It hides abstraction in a place where there is so little need for it: While I might not care how divisibility is decided, I do care that when determining leap years, the recipe operates at the abstraction level of determining divisibility.

Reply:
Hi Simon, I agree with this, the point was more of to illustrate concepts. I've deleted this section as it obviously causes confusion.